H.Ws/SQL

```diff
@@ -0,0 +1,36 @@
+show databases;
+use baseball;
+
+show tables;
+
+
+#Historic Average 
+
+SELECT (SUM(Hit)/SUM(atBat)) FROM batter_counts AS Batting_Average
+GROUP BY batter;
+
+
+#Create table with date
+
+CREATE TABLE IF NOT EXISTS Averages_table
+	AS (SELECT b.Hit, b.atBat, b.batter, b.game_id, g.local_date 
+		FROM batter_counts b 
+		JOIN game g
+		ON b.game_id = g.game_id);
+
+
+#Annual Average
+		
+SELECT (SUM(Hit)/SUM(atBat)) 
+FROM Averages_table AS Annual_Average
+GROUP BY batter & YEAR(local_date);
+
+
+#Rolling Average
+
+SELECT a.batter, a.local_date, (SUM(b.Hit)/SUM(b.atBat))
+  FROM Averages_table a
+  JOIN Averages_table b
+  	ON a.game_id = b.game_id
+ WHERE b.local_date BETWEEN a.local_date - 100 AND a.local_date
```

Comment:
I do not know the correct answer, but when i try to run your code the results are not matching with the expected results( I did a calculation on excel)

Reply:
for batter = 110029, 
date = 2012-06-24
previous 100 days = 2012-03-16
expected sum of hit. = 38
expected sum of atBat = 143
expected batter average = 0.2657

your output 

expected sum of hit. = 28
expected sum of atBat = 64
expected batter average = 0.3125

Reply:
I think its because of the fact that it shouldn't count the day you are on. 
Do you think running this would fix it:  WHERE b.local_date BETWEEN a.local_date - 100 AND a.local_date ??

Reply:
I believe it should..

Reply:
Its giving me an empty table for some reason. 
Also, is the "ON" clause working on your end when you run the code, because now its not working on my end. 
Did you have to make any changes?

Reply:
it works on my end...maybe refer to my code if that helps.

Reply:
I couldn't get it to 2.6, but very close

Reply:
I think the issue was in the ON clause. I made that to be by batter = batter, seems to fix it.

Reply:
yes batter = batter on both self joins should work.